### DISCUSSION TITLE
Block editor not saving

### DISCUSSION THREAD

**Discussion by hatefreelance** [area17/twill#1054] [Q&A]
I have a problem with the block editor: any blocks I add and save in the editor dissapear after page refresh.
But I can see them in the database blocks table, with correct entity attached, and I'm even able to render all the blocks.

Has anyone else encountered such an issue?

Twill 2.4,
PHP 7.4

My twill.php config looks like this:

```
<?php

return [
	'media_library' => [
		'disk' => 'twill_media_library',
		'endpoint_type' => env('MEDIA_LIBRARY_ENDPOINT_TYPE', 's3'),
		'cascade_delete' => env('MEDIA_LIBRARY_CASCADE_DELETE', false),
		'local_path' => env('MEDIA_LIBRARY_LOCAL_PATH', 'uploads'),
		'image_service' => env('MEDIA_LIBRARY_IMAGE_SERVICE', 'A17\Twill\Services\MediaLibrary\Glide'),
//        'acl' => env('MEDIA_LIBRARY_ACL', 'private'),
		'filesize_limit' => env('MEDIA_LIBRARY_FILESIZE_LIMIT', 50),
		'allowed_extensions' => ['svg', 'jpg', 'gif', 'png', 'jpeg', 'svg'],
		'init_alt_text_from_filename' => true,
		'prefix_uuid_with_local_path' => config('twill.file_library.prefix_uuid_with_local_path', false),
		'translated_form_fields' => true,
	],
	'glide' => [
		'base_url' => '',
		'base_path' => '/storage/uploads/'
	],
	'block_editor' => [
		'use_twill_blocks' => [],
		'block_preview_render_childs' => false,
		'crops' => [
			'full_width' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'service_cover_list' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'service_cover_detail' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'partner_logo_list' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'partner_logo_popup' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'cover' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'about_cover' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],
			'mission_cover' => [
				'default' => [
					[
						'name' => 'default',
					]
				],
			],

		]
	]
];

```

**Comment by pboivin** [comment]
Hey @hatefreelance, a bit of a long shot but did you override `getFormFields()` in your module repository? If so, did you call `parent::getFormFields()`?

**Reply by hatefreelance** [reply]
I didn't.

**Reply by hatefreelance** [reply]
I have a block named text_center. First file that is needed /resources/views/admin/blocks/text_center.blade.php - a file with form fields setting for this block. Second file is a preview blade: /resources/views/site/blocks/text_center.blade.php
Do I need to do something else to get this block working? Model  and Repository traits are also set.
And what should I put in twill.block_editor.inline_blocks_templates config? It seems that this options is what causing problems for me

**Reply by hatefreelance** [reply]
Seems like it is some strange issue with inline_blocks_templates, the moment /vendor/area17/twill/views/partials/form/utils/_blocks_templates.blade.php is being included it overrides all the values that were present in window.TWILL.STORE.form.blocks

**Reply by ifox** [reply]
Hi @hatefreelance, you should keep that configuration to the default value, `true`. Can  you clarify what you mean? Are blocks empty, or not there at all after a page refresh? Everything you shared seems correct and you shouldn't need anything to have this block working.
